### PR TITLE
bugfix: Fix typos in keyword links

### LINKED ISSUE #42213: Typos in keywords links
### Type of issue

Typo

### Description

In "Members" section, sentence "ALLOWED: Expanding the visibility of a member that is not virtual", virtual keyword link leads to sealed keyword.

In "Code Changes" section, sentence "DISALLOWED: Adding the checked keyword to a code block", checked keyword link leads to virtual keyword.

### Page URL

https://learn.microsoft.com/en-us/dotnet/core/compatibility/library-change-rules#modifications-to-the-public-contract

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/core/compatibility/library-change-rules.md

### Document Version Independent Id

79a4adc9-b2c9-68b7-669e-2bd8d2d44faf

### Article author

@gewarren

### Metadata

* ID: c4b482b0-6dd1-42de-5389-5b0b47823402 
* Service: **dotnet-fundamentals**

### PR DESCRIPTION
**Pull request by ChinoUkaegbu**
## Summary

Fixed the links for 'virtual' keyword in "Members" section and 'checked' keyword in "Code Changes" section.

Fixes #42213

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/compatibility/library-change-rules.md](https://github.com/dotnet/docs/blob/3e267e26ca0ba3fda2f84568adedc6bcbb7e8812/docs/core/compatibility/library-change-rules.md) | [docs/core/compatibility/library-change-rules](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/library-change-rules?branch=pr-en-us-42244) |

### CODE REVIEW

**Comment by ChinoUkaegbu** [comment]
Also for the `DISALLOWED: Adding the checked keyword to a code block` would it make more sense to rephrase this to `checked statement` since the docs refer to checked and unchecked as [statements](https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/statements/checked-and-unchecked.md) and not keyword

**Comment by gewarren** [comment]
> Also for the `DISALLOWED: Adding the checked keyword to a code block` would it make more sense to rephrase this to `checked statement` since the docs refer to checked and unchecked as [statements](https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/statements/checked-and-unchecked.md) and not keyword

Yes, please!

**Comment by ChinoUkaegbu** [comment]
Done!

File: docs/core/compatibility/library-change-rules.md [hunk 1]
```diff
@@ -314,7 +314,7 @@ Changes in this category modify the public surface area of a type. Most of the c
 
 - ❌ **DISALLOWED: Changing a [struct](../../csharp/language-reference/builtin-types/struct.md) to a [class](../../csharp/language-reference/keywords/class.md) and vice versa**
 
-- ❌ **DISALLOWED: Adding the [checked](../../csharp/language-reference/keywords/virtual.md) keyword to a code block**
+- ❌ **DISALLOWED: Adding the [checked](../../csharp/language-reference/statements/checked-and-unchecked.md) statement to a code block**
 
    This change may cause code that previously executed to throw an <xref:System.OverflowException> and is unacceptable.
```
Comment (gewarren) [inline docs/core/compatibility/library-change-rules.md:319]:
```suggestion
   This change might cause code that previously executed to throw an <xref:System.OverflowException> and is unacceptable.
```

**Review by gewarren — APPROVED** [review]

**Comment by gewarren** [comment]
Thanks @ChinoUkaegbu!
